doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1265 @@
+
+id              : Unique sequential ID generated by the system for every event {uint64}
+type-id         : Name of the event generated {string}
+text            : Dynamic message describing the cause for the event {string}
+time-created    : Time stamp at which the event is generated {uint64}
+action          : Indicates action of the event; for one-shot events, it is empty. For alarms it could be raise, clear or acknowledge {enum}
+resource        : Object which generated the event {string}
+severity        : Severity of the event {string}
+
+127.0.0.1:6379[6]> hgetall "EVENT|1"
+ 1) "text"
+ 2) ":- handle_custom_evprofile: Custom Event Profile x.json is applied."
+ 3) "type-id"
+ 4) "CUSTOM_EVPROFILE_CHANGE"
+ 5) "id"
+ 6) "1"
+ 7) "time-created"
+ 8) "1621459327118629520"
+ 9) "resource"
+10) "/etc/evprofile/x.json"
+11) "severity"
+12) "WARNING"
+127.0.0.1:6379[6]>
+```
+
+Schema for EVENT_STATS table is as follows:
+```
+EVENT_STATS Table:
+==============================
+
+Key             : id
+
+id              : key {state}
+events          : Total events raised {uint64}
+raised          : Total alarms raised {uint64}
+cleared         : Total alarms cleared {uint64}
+acked           : Total alarms acknowledged {uint64}
+
+127.0.0.1:6379[6]> hgetall "EVENT_STATS|state"
+1) "events"
+2) "1"
+3) "raised"
+4) "0"
+5) "cleared"
+6) "0"
+7) "acked"
+8) "0"
+127.0.0.1:6379[6]>
+```
+Current Alarm Table will not have any limits as it only contains the snapshot of the alarms during the current run.
+
+Contents of an alarm record. In this case, the alarm was raised temperature crossed a threshold.
+```
+ALARM Table:
+==============================
+
+Key             : id
+
+id              : Unique sequential ID generated by the system for every event {uint64}
+type-id         : Name of the event generated {string}
+text            : Dynamic message describing the cause for the event {string}
+time-created    : Time stamp at which the event is generated {uint64}
+acknowledged    : Indicates if alarm has been acknowledged {boolean}
+resource        : Object which generated the event {string}
+severity        : Severity of the event {string}
+acknowledged    : Indicates when alarm has been acknowledged/unacknowledged {uint64}
+
+127.0.0.1:6379[6]> hgetall "ALARM|2"
+ 1) "type-id"
+ 2) "TEMPERATURE_EXCEEDED"
+ 3) "text"
+ 4) "temperatureCrossedThreshold: Current temperature for sensor/2 is 76 degrees"
+ 5) "action"
+ 6) "RAISE"
+ 7) "resource"
+ 8) "sensor/2"
+ 9) "time-created"
+10) "1621460371062299951"
+11) "severity"
+12) "CRITICAL"
+13) "id"
+14) "2"
+15) "acknowledged"
+16) "false"
+127.0.0.1:6379[6]
+```
+
+Schema for ALARM_STATS table is as below. When an alarm of particular severity is cleared,
+the corresponding severity counter is decremented. 
+```
+ALARM_STATS Table:
+==============================
+
+Key                       : id
+
+id                        : key {state}
+alarms                    : Number of active alarms {uint64}
```

Comment:
what does active means? cleared/acknowledged?

Reply:
An Active alarm is the one that is not cleared yet by the application.